data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/network_action/PstSetpoint.java

```diff
@@ -63,15 +82,20 @@ public PstSetpoint(String id, NetworkElement networkElement, double setpoint) {
     @Override
     public void apply(Network network) {
         PhaseTapChanger phaseTapChanger = network.getTwoWindingsTransformer(networkElement.getId()).getPhaseTapChanger();
-        if (phaseTapChanger.getHighTapPosition() - phaseTapChanger.getLowTapPosition() + 1 >= setpoint && setpoint >= 1) {
+
+        if (rangeDefinition.equals(CENTERED_ON_ZERO)
+                && setpoint > phaseTapChanger.getLowTapPosition() && setpoint < phaseTapChanger.getHighTapPosition()) {
+            phaseTapChanger.setTapPosition((int) setpoint);
+        } else if (rangeDefinition.equals(STARTS_AT_ONE)
+                && (phaseTapChanger.getHighTapPosition() - phaseTapChanger.getLowTapPosition() + 1 >= setpoint && setpoint >= 1)) {
             phaseTapChanger.setTapPosition(phaseTapChanger.getLowTapPosition() + (int) setpoint - 1);
         } else {
```

Comment:
Here we assume, that the PowSyBl phase tap changer is always centered on zero which I'm not sure is always correct

Reply:
is there a way to know which convention is used in the network ?

Reply:
is there a way to know which convention is used in the network ?

Reply:
Its done